File: Palindrome.c
bool isPalindrome(int x)
{   
    if(x<0)
    {
        return false;
    }

    int val = x;
    int size = 0;
    
    while(val/10 != 0)
    {
        size++;
        val = val/10;
    }

    ++size;
    
    int first_element = x % 10;
    int val_array[size];
    val_array[0] = first_element;

    for(int i = 1; i<size; i++)
    {
        val_array[i] = (int) (x /  pow(10, i)) % 10;
    }

    int compare = sizeof(val_array)/sizeof(val_array[0]) - 1;
    int count = 0;

    while(count != compare)
    {   

        if(val_array[count] == val_array[compare])
        {
            if(compare - 1 == count)
            {
                return true;
            }
            count++;
            compare--;
        }
        else
        {
            return false;
        }
    }
       
    
    return true;
}

File: Palindrome.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x<0:
            return False
        
        x_list = [int(i) for i in str(x)]
        
        while(x_list):
            if(x_list[0] == x_list[len(x_list)-1]):
                x_list = x_list[1:-1]
            else:
                return False


        return True